### PR TITLE
WIP: Prevent CPU lockup in servo_cpp_interface_demo

### PR DESCRIPTION
**Pull request by AndyZe**
This addresses issue #268 for one executable. If everybody's amenable, there are some other places this type of change might help:

- test/servo_launch_test_common.hpp
- test/test_servo_parameters.cpp
- test/test_servo_parameters_node.cpp
- test/publish_fake_jog_commands.cpp

### CODE REVIEW

File: moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp [hunk 1]
```diff
@@ -186,8 +187,12 @@ int main(int argc, char** argv)
 
   // Spin
   auto executor = std::make_unique<rclcpp::executors::MultiThreadedExecutor>();
-  executor->add_node(node);
-  executor->spin();
+  rclcpp::Rate loop_rate(DEFAULT_SPIN_RATE);
+  while (rclcpp::ok())
+  {
+    executor->spin_node_once(node);
```
Comment (henningkayser) [inline moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp:193]: Why not use spin_some()?

Comment (AndyZe) [reply moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp:193]: I'm not aware of the pros/cons of that. Do you have any reason to suggest it? I will try it.

Comment (AndyZe) [reply moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp:193]: I find that replacing `executor->spin_once(node);` with `executor->spin_some();` does not work -- the robot doesn't move and CPU load is just as high.

Comment (AndyZe) [reply moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp:193]: Oh, it does work if I do it right. Seems to be functionally equivalent to spin_once() as far as I can tell

**Comment by AndyZe** [comment]
I'm going to add this in other places where it seems useful now.

File: moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp (hunk 1 above)
Comment (mikeferguson) [reply moveit_ros/moveit_servo/src/cpp_interface_demo/servo_cpp_interface_demo.cpp:193]: I think the advantage of spin_some() will be when you have multiple nodes on an executor (and possibly some small gains from not adding the node to the executor and then removing it at the end of each call to spin once).

**Comment by AndyZe** [comment]
Marking as WIP until root cause is sorted out better.

**Pull request closed without merging**